src/components/AuthForm.js

```diff
@@ -0,0 +1,142 @@
+import {
+  Form,
+  Link,
+  useSearchParams,
+  useActionData,
+  useNavigation,
+  useNavigate,
+} from "react-router-dom";
+
+import classes from "./AuthForm.module.css";
+import { useState } from "react";
+
+function AuthForm() {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const isLogin = searchParams.get("mode") === "login";
+  const [otp, setOtp] = useState(false);
+  const navigation = useNavigation();
+  const isSubmitting = navigation.state === "submitting";
+  const data = useActionData();
+  const navigate = useNavigate();
+
+  async function submitHandler(event) {
+    event.preventDefault();
+    if (!otp) {
+      const data = new FormData(event.target);
+      var object = {};
+      data.forEach((value, key) => (object[key] = value));
+      var body = JSON.stringify(object);
+
+      let url = "http://127.0.0.1:8000/user/";
+
+      if (isLogin) {
+        url += "login/";
+      } else {
+        url += "register_user/";
+      }
```

Comment:
I think it could made more simple by
 `url += isLogin ? "login/" : "register_user/";`